src/main/java/htsjdk/samtools/util/IntervalList.java

```diff
@@ -539,72 +527,17 @@ public static IntervalList fromReader(final BufferedReader in) {
             final IntervalList list = new IntervalList(codec.decode(headerReader, "BufferedReader"));
             final SAMSequenceDictionary dict = list.getHeader().getSequenceDictionary();
 
-            //there might not be any lines after the header, in which case we should return an empty list
+            // There might not be any lines after the header, in which case we should return an empty list
             if (line == null) {
                 return list;
             }
 
+            final IntervalListCodec intervalListCodec = new IntervalListCodec(dict);
```

Comment:
Ideally `IntervalListCodec` wouldn't have to provide this  constructor, since the codec should really get the dictionary from it's input stream directly. Also, the code above that parses out the header lines is redundant with code thats already in the codec; the codec implementation is better because it doesn't require reading the entire header into a string that is then re-parsed. But, I don't see a way to bridge from the `BufferedReader` to the codec. We might want to think about deprecating `fromReader(final BufferedReader in)`; it would be easy to replace it with a (stream/file/path) overload that could use the codec directly.

Reply:
I don't think that there's an action item here...please correct me if I'm wrong :-)